core/tee/tee_svc_storage.c

```diff
@@ -543,6 +546,19 @@ TEE_Result syscall_storage_obj_del(unsigned long obj)
 	if (o->pobj == NULL || o->pobj->obj_id == NULL)
 		return TEE_ERROR_BAD_STATE;
 
+	if (o->info.handleFlags & TEE_DATA_FLAG_SHARE_CRYPTO) {
```

Comment:
not sure what you mean by proprietary: it a new generic flag indicating that the buffer is to be shared with the crypto layer before deletion.

the way things work at the moment, given the number of layers that will need to be modified, is that all ECC/RSA keys generated by the se050 driver are always persistent in NVM by definition. So when the TA creates the keys - since it knows the type of key being created wheter it is rsa, ecc or whatever - it knows whether to set the flag or not.

This flag (or similar) should clearly be added to the crypto key generation request but I thought it would be easier to upstream if the driver initially assumed that all the keys that it generates are persistent.

The driver of course can also work with user provided keys btw.

Reply:
> not sure what you mean by proprietary: it a new generic flag indicating that the buffer is to be shared with the crypto layer before deletion.

proprietary == not standard. The GP spec does not document implementation defined flags in `handleFlags` so if GP decides to use this bit for something else we're screwed.

> the way things work at the moment, given the number of layers that will need to be modified, is that all ECC/RSA keys generated by the se050 driver are always persistent in NVM by definition.

I'm fine with that.

>  So when the TA creates the keys - since it knows the type of key being created wheter it is rsa, ecc or whatever - it knows whether to set the flag or not.

The TEE core knows that also, it does not need the TA to tell it.

Reply:
ok